packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -2098,10 +2099,37 @@ export async function getInputConnectionData(
 
 		if (!connectedNodeType.supplyData) {
 			if (connectedNodeType.description.outputs.includes(NodeConnectionType.AiTool)) {
+				let toolRunIndex = 0;
```

Comment:
Could you please add a comment here explaining how `runIndex` and `toolRunIndex` 🙏 It's a bit hard to parse at a first glance 
```suggestion
// toolRunIndex keeps track of how many times this specific AI tool node has been executed
// It's incremented each time the tool is invoked, allowing us to track separate executions 
let toolRunIndex = 0;
				let toolRunIndex = 0;
```